src/target/ext/host_io.rs

```diff
@@ -365,12 +357,11 @@ pub trait HostIoSetfs: HostIo {
     /// remote targets where the remote stub does not share a common filesystem
     /// with the inferior(s).
     ///
-    /// If pid is nonzero, select the filesystem as seen by process pid. If pid
-    /// is zero, select the filesystem as seen by the remote stub.
+    /// See [`FsKind`] for the meaning of argument.
     ///
-    /// Return 0 on success, or -1 if an error occurs. If vFile:setfs: indicates
-    /// success, the selected filesystem remains selected until the next
-    /// successful vFile:setfs: operation.
+    /// Return 0 on success, or [`HostIoError::Errno`] if an error occurs. If
+    /// setfs indicates success, the selected filesystem remains selected
+    /// until the next successful setfs operation.
     fn setfs(&mut self, fs: FsKind) -> HostIoResult<u32, Self>;
```

Comment:
the spec stipulates that setfs will only ever return 0 (on success), or -1 + errno (on error).

as such, the return type here should be `HostIoResult<(), Self>`, where `Ok(())` is returned on success.